### ISSUE TITLE
Memory leak of gRPC server

### ISSUE THREAD

**Issue by Allkoman** [grpc/grpc-go#3178]
Please see the FAQ in our main README.md, then answer the questions below before
submitting your issue.

### What version of gRPC are you using?
v1.25.0
### What version of Go are you using (`go version`)?
1.11.0
### What operating system (Linux, Windows, …) and version?
k8s golang:latest 
### What did you do?
If possible, provide a recipe for reproducing the error.
I'm using golang grpc as a server and my memory will just rise and never fall until the pods die
[top3 goroutine](https://ftp.bmp.ovh/imgs/2019/11/609d53a6ce910528.jpg)
[web goroutine](https://ftp.bmp.ovh/imgs/2019/11/31ee3b5d5870e459.jpg)
[web heap](https://ftp.bmp.ovh/imgs/2019/11/c8b5a791780cac24.jpg)
[top1 heap 99%](https://ftp.bmp.ovh/imgs/2019/11/3bd5b1167682bb78.jpg)

maybe some problem here?
```
func (s *Server) serveStreams(st transport.ServerTransport) {
	defer st.Close()
	var wg sync.WaitGroup
	st.HandleStreams(func(stream *transport.Stream) {
		wg.Add(1)
		go func() {
			defer wg.Done()
			s.handleStream(st, stream, s.traceInfo(st, stream))
		}()
	}, func(ctx context.Context, method string) context.Context {
		if !EnableTracing {
			return ctx
		}
		tr := trace.New("grpc.Recv."+methodFamily(method), method)
		return trace.NewContext(ctx, tr)
	})
	wg.Wait()
}
```

### What did you see instead?
normal memory , thx very much

**Comment by dfawley** [comment]
We need more information to assist here.  Ideally a sample program and instructions for how to run it to make the leak happen.  The memory numbers I can see are single-digit megabytes which does not seem excessive or indicative of a leak.

**Comment by LucasDove** [comment]
came across this problem, too. since we are using grpc-go v1.19.0，it seems that this problem has not been solved from at least this version to v1.25.0
